torch/_refs/__init__.py

```diff
@@ -520,6 +520,50 @@ def log10(a):
     return prims.log10(a)
 
 
+@out_wrapper
+def log_softmax(
+    a: TensorLikeType,
+    dim: int,
+    *,
+    dtype: Optional[torch.dtype] = None,
+) -> TensorLikeType:
+    result_dtype = dtype or a.dtype
+    computation_dtype = utils.get_computation_dtype(a.dtype)
+    a = prims.convert_element_type(a, computation_dtype)
+    a_max = amax(a, dim, keepdim=True)
+    shifted = a - a_max
+    shifted_logsumexp = logsumexp(shifted, dim, keepdim=True)
+    return prims.convert_element_type(
+        shifted - shifted_logsumexp, result_dtype
+    )
+
+
+def _squeeze_multiple(a: TensorLikeType, dims: DimsSequenceType) -> TensorLikeType:
+    for dim in reversed(dims):
+        a = squeeze(a, dim)
+    return a
+
+
+@out_wrapper
+def logsumexp(
+    a: TensorLikeType,
+    dims: DimsType,
```

Comment:
"dim"? 

https://pytorch.org/docs/master/generated/torch.logsumexp.html?highlight=logsumexp#torch.logsumexp

Reply:
Right, it should be dim to match the torch namespace. I think I was confused because it also accepts and works for several dimensions
> dim (int or tuple of python:ints)
```py
import torch
a = torch.ones(3, 3)
torch.logsumexp(a, (0, 1))
# tensor(3.1972)
```